scripts/s3manage.py

```diff
@@ -0,0 +1,568 @@
+
+def get_version_info(version):
+    obj = s3.Object(bucket.name, '{}/info.json'.format(version))
+    try:
+        content = obj.get()["Body"].read()
+        raw = _unzip_data(content)
+        data = json.loads(raw)
+    except botocore.exceptions.BotoCoreError:
+        return None
+    return data
+
+
+def list_version():
+    active_version = int(get_active_version())
+
+    indexes = bucket.objects.filter(Prefix="index").all()
+
+    p = re.compile(ur'index.(\d+).html')
+    print "Version      Build date"
+    print "-----------+------------------------"
+    for index in indexes:
+        match = re.search(p, index.key)
+        if match:
+            version = int(match.groups()[0])
+            print version, index.last_modified, 'active' if version == active_version else ''
+
+
+def version_info(version):
+    info = get_version_info(version)
+    if info is None:
+        print "No info for version {}".format(version)
+        sys.exit(1)
+    for k in info.keys():
+        print "{}: {}".format(k, info[k])
+
+
+def delete_version(version):
+    if version_exists(version) is False:
+        print("Version '{}' does not exists in AWS S3. Aborting".format(version))
+        sys.exit(2)
+
+    if version == get_active_version():
+        print("Version '{}' is the active version. You cannot delete it".format(version))
+        sys.exit()
+
+    files = bucket.objects.filter(Prefix=str(version)).all()
+
+    indexes = [{'Key': k.key} for k in files]
+    for n in ('index', 'embed', 'mobile'):
+        src_key_name = '{}.{}.html'.format(n, version)
+        indexes.append({'Key': src_key_name})
+
+    resp = s3client.delete_objects(Bucket=BUCKET_NAME, Delete={'Objects': indexes})
+
+    for v in resp['Deleted']:
+        print v
+
+
+def activate(version):
+    if version_exists(version) is False:
+        print("Version '{}' does not exists in AWS S3. Aborting".format(version))
+        sys.exit(2)
+
+    if version == get_active_version():
+        print("Version '{}' is already the active version. Doing nothing".format(version))
+        sys.exit()
+
+    # Source files
+
+    files = bucket.objects.filter(Prefix='{}/src'.format(version)).all()
+
+    for k in files:
+        src_key_name = k.key
+        dst_key_name = src_key_name.replace('{}/'.format(version), '')
+        print("{} --> {}".format(src_key_name, dst_key_name))
+
+        s3client.copy_object(
+            Bucket=BUCKET_NAME,
+            CopySource=BUCKET_NAME +
+            '/' +
+            src_key_name,
+            Key=dst_key_name,
+            ACL='public-read')
+
+    # Prod files
+    for n in ('index', 'embed', 'mobile'):
+        src_key_name = '{}.{}.html'.format(n, version)
+        print("{} --> {}.html".format(src_key_name, n))
+
+        s3client.copy_object(
+            Bucket=BUCKET_NAME,
+            CopySource=BUCKET_NAME +
+            '/' +
+            src_key_name,
+            Key=n +
+            '.html',
+            ACL='public-read')
```

Comment:
ok nice, I understand what is being done here. Only move 3 files and the app is deployed. in prod. I would use only one function to do both operations though.